oxidation/libparsec/crates/platform_device_loader/src/native/local_device_file.rs

```diff
@@ -0,0 +1,278 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+use std::{
+    collections::HashSet,
+    ffi::OsStr,
+    fs::File,
+    path::{Path, PathBuf},
+};
+
+use libparsec_client_types::{
+    AvailableDevice, DeviceFile, DeviceFilePassword, DeviceFileRecovery, DeviceFileType,
+    LocalDevice, LocalDeviceError, LocalDeviceResult,
+};
+use libparsec_crypto::SecretKey;
+
+/// TODO: Remove me once wasm doesn't need to mock
+pub async fn test_gen_default_devices() {
+    panic!("This function shouldn't be called in native platform")
+}
+
+async fn _list_available_devices(config_dir: &Path) -> LocalDeviceResult<Vec<AvailableDevice>> {
+    let mut list = vec![];
+    // Set of seen slugs
+    let mut seen = HashSet::new();
+
+    let key_file_paths = config_dir.join("devices");
+
+    // Consider `.keys` files in devices directory
+    let mut key_file_paths = read_key_file_paths(key_file_paths)?;
+
+    // Sort paths so the discovery order is deterministic
+    // In the case of duplicate files, that means only the first discovered device is considered
+    key_file_paths.sort();
+
+    for key_file_path in key_file_paths {
+        let device = match load_available_device(key_file_path) {
+            // Load the device file
+            Ok(device) => device,
+            // Ignore invalid files
+            Err(_) => continue,
```

Comment:
Are you handling `LegacyDeviceFile`?

Reply:
I don't know, if you changed anything, I can include in

Reply:
I refactored `legacy_device_loading` in its own function look at #3896. Isn't native platform outside the scope of this PR? I can rebase my on going PR with the new architecture on this PR has been merged

Reply:
This PR migrate some `native` stuff and implement `list_available_devices` for the web platform

Reply:
Done